megamek/src/megamek/client/ui/swing/BoardEditor.java

```diff
@@ -152,7 +151,7 @@ public int compareTo(TerrainHelper o) {
         @Override
         public boolean equals(Object other) {
             if (other instanceof Integer) {
-                return getTerrainType() == ((Integer)other).intValue();
+                return getTerrainType() == (Integer) other;
```

Comment:
Why would it only work between those values? Integer is just an int wrapper class, so this is an int to int comparison which should work between -2^-31 and 2^31 - 1.
I've also not found any uses of this in the code, but I believe the reasoning behind it is that this is supposed to be used in terrainType comparisons

Reply:
That isn't a comparison of `int`, that's a comparison of `Integer` by reference rather than value. `Integer` values between -128 and 127 *tend* to be the same reference, but other values will have unique references.

You need to use `Integer.equals(val1, val2)` or `val1.equals(val2)` to compare values.

Reply:
@sixlettervariables 
Unless my testing is incorrect, the caching is only included for Integer to Integer comparisons. This is an int to Integer comparison that will automatically unbox properly. I can share my test code if you'd like to check my math (found it an interesting conundrum and ended up going down the rabbit hole).

Reply:
This was highlighted as an issue when I used SpotBugs, so I'd rather make it explicit we want value comparisons.

Reply:
Funny thing is that I made this change for the exact same reason, the Idea IDE pointed this out as a potential bug and this was the suggested fix 😂

Reply:
I guess the different tools have different ways of avoiding the issue. If that was IntelliJ's suggested fix that's fine by me.

Reply:
For reference: Integer caches wrapper instances for all values between -128 and 127 and uses a single instance for each of those values no matter where they come from. It also caches any instance created using the `valueOf` static method for reuse, but not those created with the Integer(int) constructor, which is why that has been deprecated in Java 9. The same is true of the other wrapper classes.

So for comparisons of two Integer objects, == will have the desired outcome if you can guarantee they both within the automatic caching range or were created using valueOf, but equals is preferred as it will work regardless.

In the case of comparison between int and Integer, as in this case, I suspect that `other.equals(getTerrainType())`, would be less efficient, since it would result in automatic boxing rather than unboxing. Though the compiler may be clever enough to recognize that and use unboxing instead.